subprojects/core-api/src/main/java/org/gradle/api/tasks/util/PatternSet.java

```diff
@@ -174,6 +176,19 @@ public Spec<FileTreeElement> getAsExcludeSpec() {
         return patternSpecFactory.createExcludeSpec(this);
     }
 
+    /**
+     * Like {@link #getIncludes()}, but returns a view or empty set.
+     *
+     * <p>Use this if you are only reading from the set, as it avoids allocating a set if not needed.</p>
+     *
+     * @return the include patterns, or an empty set if none
+     * @since 8.8
+     */
+    @Incubating
+    public Set<String> getIncludesView() {
```

Comment:
💭 The suffix `View` doesn't really communicate that the result is immutable to me. 
 `includesAsImmutable()` might be more in line with other parts of our API (I'm thinking of attributes).  Also, could the immutability of the results for these methods be noted in the javadoc?

Reply:
It's not immutable -- it's just unmodifiable, as noted by Leonard above. Does the name make sense with that context?

Reply:
Oh, I missed his comment.  Then maybe just note that explicitly in the javadoc then.  When I read `view` I don't get that - some views are mutable.